### ISSUE TITLE
Question about updating an entity that triggers a rule

### ISSUE THREAD

**Issue by eyeonall** [CiviCooP/org.civicoop.civirules#198]
First, this extension is really really nice. I have been looking for a way to help with an upgrade to D8 and migrating the business logic contained in the civicrm entity D7 rules into this extension looks very promising.

The question I have has to do with the triggering of the rules from the hook_civicrm_post. The Update Activity Status example changes the status of the activity. The Notes section of the CiviCRM developer docs for the post hook say:
"These hooks use database transactions. Don't execute code that updates the same data in the database without using a callback. Eg. if triggering on a Membership entity, don't try and update that membership entity within the hook. Use CRM_Core_Transaction::addCallback() instead."

I have been digging through the code but, I am embarrassed to say, my php OOP skills are not nearly as well honed as yours so I cannot tell how the activity gets updated without altering the same data in the database. I do not see a callback. Thing is, the rule works so I have to assume that this was taken into account and the transaction is not broken by the rule action being triggered in the post hook as the dev docs would indicate.

Does that mean I can update anything in an entity that triggered the rule without needing a callback?

Thanks again and I am sorry for being a pain. 

A.J.

**Comment by jaapjansma** [comment]
The answer is yes you can. However if you have a trigger based on activity changed. And you update activity data the trigger is called again. 

If your condition has something if status = scheduled and your action is update status to completed then the rule is safe to do. As it doesnt get executed twice.

**Comment by eyeonall** [comment]
Yes that is definitely logical. How about something like an address 
location type id. We have a membership of a lot of old folks who spend 
the summers is warmer climates like Florida or Arizona. Turns out I am 
also using your company's future address extension to deal with this. 
Had to make some minor adjustments to get it to install and work with 
4.7 and it does the job.

To make it simple for the chapter secretaries, I am only using the 
future temporary address. For those 'snow birds' I added a custom field 
which allows the chapter secretaries to mark the temporary address as a 
'permanent' temporary address. When the cron runs and rotates the temp 
address out it deletes it. A post hook exists, soon to be a rule, that 
takes the deleted address, checks to see if it was a permanent one, and 
recreates it as an 'Other' that can be set as a temp address at a later 
date.

For new_, since I am only using temp_, I want to catch it, check if 
there is a return date, and change the location type to new_ is there is 
none. I just didnt want to chance breaking the transaction by making 
that type of change.



------ Original Message ------
